// Copyright 2025 The SiliFuzz Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto2";

package silifuzz.proto;

import "google/protobuf/timestamp.proto";

message HashTestResult {
  enum Status {
    // This should never occur, included so we absolutely notice a missing or
    // mangled message.
    UNINITIALIZED = 0;
    // The test completed successfully.
    OK = 1;
    // Data corruption was detected.
    FAILED = 2;
    // This particular CPU platform is currently not supported.
    PLATFORM_NOT_SUPPORTED = 3;
  }

  // The hostname of the machine the test ran on.
  optional string hostname = 1;
  // The platform the test generated instructions for.
  // This may not be the same as the actual platform of the machine.
  optional string platform = 2;
  // The version of hashtest that was run.
  optional string version = 3;

  // Is this machine good or bad, or did something go wrong trying to test it?
  optional Status status = 4;

  // The beginning and end of the testing period.
  optional google.protobuf.Timestamp testing_started = 5;
  optional google.protobuf.Timestamp testing_ended = 6;

  // Test statistics.
  optional uint64 tests_run = 7;
  optional uint64 tests_failed = 8;

  // CPU statistics.
  repeated uint32 tested_cpus = 9 [packed = true];
  repeated uint32 suspected_cpus = 10 [packed = true];
}
